.github/actions/generate-commit-list-v1/src/run.test.ts

```diff
@@ -70,14 +68,63 @@ describe('run', () => {
 
       assert.isTrue(setFailed.calledOnce)
       assert.isTrue(
-        setFailed.calledWith('Input required and not supplied: head')
+        setFailed.calledWith(
+          'You must provide either a tag or both a base and head branch.'
+        )
       )
     })
   })
 
+  describe('when the tag input is provided', () => {
+    describe('and the base input is provided', () => {
+      it('should throw an error', async () => {
+        getInput.withArgs('base').returns('main')
+        getInput.withArgs('head').returns('')
+        getInput.withArgs('tag').returns('v1.0.0')
+
+        const core = {
+          getInput,
+          setFailed
+        }
+
+        await run(core as unknown as Core, github as unknown as Github)
+
+        assert.isTrue(setFailed.calledOnce)
+        assert.isTrue(
+          setFailed.calledWith(
+            'You cannot provide both a tag and both a base and head branch.'
+          )
+        )
+      })
+    })
+
+    describe('and the head input is provided', () => {
+      it('should throw an error', async () => {
```

Comment:
Should have another test to verify when both `head` and `base` are provided

Reply:
we have 3 test cases. Do I have to add something else?

https://github.com/dequelabs/axe-api-team-public/blob/feat-add-tags-commmit-list/.github/actions/generate-commit-list-v1/src/run.test.ts#L124-L242